src/config.c

```diff
@@ -7,21 +7,101 @@
 
 #include "consts.h"
 #include "module.h"
+#include "parse_policies.h"
 #include "query_language.h"
 #include "RedisModulesSDK/redismodule.h"
 
 #include <assert.h>
 #include <string.h>
+#include <float.h>
 #include "rmutil/strings.h"
 #include "rmutil/util.h"
 
+/*
+ * Logs that a deprecated configuration option was used.
+ */
+#define LOG_DEPRECATED_OPTION(deprecatedName, modernName, show)                                    \
+    if (show) {                                                                                    \
+        RedisModule_Log(rts_staticCtx,                                                             \
+                        "warning",                                                                 \
+                        "%s is deprecated, please use the '%s' instead",                           \
+                        deprecatedName,                                                            \
+                        modernName);                                                               \
+    }
+
+#define LOG_DEPRECATED_OPTION_ALWAYS(deprecatedName, modernName)                                   \
+    LOG_DEPRECATED_OPTION(deprecatedName, modernName, true)
+
 TSConfig TSGlobalConfig;
 
+static RedisModuleString *tempStringStringToJustReturnInTheGetter = NULL;
```

Comment:
I know the API is obnoxious, but can we please use less obnoxious variable names despite that?

Reply:
Can you suggest a name?

Reply:
maybe something like `getConfigStringCache`